diff --git a/GNUmakefile b/GNUmakefile
index 2c6716a..9249c80 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -21,9 +21,9 @@ $(TOPDIR)/Makeconf:
 	$(error Makeconf not found, please run ./configure.sh)
 include Makefile.common
 
-SUBDIRS := bindings tenders elftool tests
+SUBDIRS := bindings tenders tests
 
-tests: bindings elftool
+tests: bindings
 
 .PHONY: $(SUBDIRS)
 
diff --git a/tests/Makefile.tests b/tests/Makefile.tests
index d4e1191..45560a8 100644
--- a/tests/Makefile.tests
+++ b/tests/Makefile.tests
@@ -34,15 +34,12 @@ BINDINGS.muen := $(TOPDIR)/bindings/muen/solo5_muen.o
 LDS.genode := $(TOPDIR)/bindings/genode/genode_dyn.ld
 BINDINGS.genode := $(TOPDIR)/bindings/genode/solo5.lib.so
 
-ELFTOOL := $(TOPDIR)/elftool/solo5-elftool
-
 # As a test is currently a single source file, we don't try to do any 
 # dependency auto-generation here, and just hard-code all expected dependencies
 # as pre-requisites below.
 
-manifest.c: manifest.json ../../include/solo5/mft_abi.h $(ELFTOOL)
-	@echo "ELFTOOL $<"
-	$(ELFTOOL) gen-manifest $< $@
+manifest.c: manifest.json ../../include/solo5/mft_abi.h
+	solo5-elftool gen-manifest $< $@
 
 %.o: %.c ../../include/solo5/solo5.h
 	@echo "CC $<"
