diff --git a/bindings/genode/bindings.cc b/bindings/genode/bindings.cc
index a4a4a34..1019a39 100644
--- a/bindings/genode/bindings.cc
+++ b/bindings/genode/bindings.cc
@@ -475,8 +475,8 @@ struct Solo5::Platform
 	net_acquire(const char *name, solo5_handle_t &handle, solo5_net_info &info)
 	{
 		unsigned index = ~0;
-		struct mft_entry *me = mft_get_by_name(&mft, name, MFT_DEV_NET_BASIC, &index); {
-		if (me != nullptr && index < MFT_MAX_ENTRIES)
+		mft_entry *me = mft_get_by_name(&mft, name, MFT_DEV_NET_BASIC, &index);
+		if (me != nullptr && index < MFT_MAX_ENTRIES) {
 			handle = index;
 			return devices[index]->net_info(info);
 		}
